crates/secret-service-client/src/musig2.rs

```diff
@@ -159,13 +159,12 @@ impl Musig2SignerFirstRound<Client, Musig2SecondRound> for Musig2FirstRound {
 
     async fn receive_pub_nonces(
         &mut self,
-        nonces: HashMap<XOnlyPublicKey, PubNonce>,
+        nonces: impl Iterator<Item = (XOnlyPublicKey, PubNonce)> + Send,
```

Comment:
since you're serializing them anyway don't you only need an iterator of borrowed items? For the server the owned ones make sense I think. Does your trait system make it such that it's required for us to make the types the same on the client and server? I don't think the extra clone is an issue fwiw but I've just been paying more attention to owned vs borrowed stuff lately and I'm curious.

Reply:
It's not super important. Don't worry about it. Just something to let simmer in the back of your mind.